src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A Stream abstraction over the TDS protocol.
+    /// The stream can be used to read and write TDS messages on a 
+    /// SQL Server physical connection.
+    /// </summary>
+    internal class TdsStream : Stream
+    {
+        public override bool CanRead => throw new NotImplementedException();
```

Comment:
I think the `Can*` calls can simply chain into the underlying stream (and add an `<inheritdoc />`

Reply:
But I suspect these will be part of a later PR?

Reply:
Yeah, I will add the `<inheritdoc />` 
And the implementation is going to evolve where this TdsStream is composed of TdsReadStream and TdsWriteStream. 
The TdsStream is what the rest of the driver will use and it will delegate calls to Read and WriteStreams internally. 

Many of the Can* methods will simply return true. e.g. CanRead, CanWrite, because in TDS we dont know whether we can read or write, till we actually Read / Write. The implementor of the protocol need to make sure that they are not reading / writing when they are not supposed to.